src/robot/libraries/Process.py

```diff
@@ -524,8 +526,37 @@ def _manage_process_timeout(self, handle, on_timeout):
 
     def _wait(self, process):
         result = self._results[process]
-        result.rc = process.wait() or 0
-        result.close_streams()
+
+        _take_stdout = (process.stdout and (not process.stdout.closed))
+        _take_stderr = (process.stderr and (not process.stderr.closed))
+
+        # This is needed to handle stdin closes identically to how it was when the
+        # wait function was used.
+        _stdin = process.stdin
+        process.stdin = process.stdin if (process.stdin and not process.stdin.closed) else io.StringIO("")
```

Comment:
It's true that `Popen.communicate()` doesn't like `stdin` to be a closed PIPE, but there can also be problems if `stdout` and `stderr` are closed PIPEs. How they are all handled is pretty inconsistent and I submitted a bug about that:
https://github.com/python/cpython/issues/131064

To be consistent with our old `Popen.wait()` based code, we need to handle all streams. Closed streams can be replaced with `None` instead `io.StringIO("")`, though.